src/client/preview/client_api.js

```diff
@@ -67,10 +99,10 @@ export default class ClientApi {
             return decorated(context);
           }, context);
         };
-      }, getStory);
+      }, storyFn);
 
       // Add the fully decorated getStory function.
-      this._storyStore.addStory(kind, storyName, fn);
+      this._storyStore.addStory(kindConfig, storyName, { storyFn: fn, storyMeta });
```

Comment:
Agree other addons are important things. We certainly have to ensure backward compatibility. I was just wondering, how do you plan to pass meta data with addons like `addon-info`? :)